doc/md/12-garbage-collection.md

```diff
@@ -0,0 +1,46 @@
+---
+sidebar_position: 12
+---
+
+# Garbage collection
+
+Garbage collection is an automatic process that manages memory usage by removing unreferenced or dead objects and freeing up allocated heap memory.
+
+Motoko’s default garbage collection method uses a copying approach that depends on the amount of heap memory currently in use. An alternative garbage collector uses a marking approach, which instead considers the amount of free heap memory. These garbage collectors are triggered when enough changes have been made to heap memory since the last collection cycle. Garbage collection can be forced to run after every message using the `--force-gc` flag in the project's `dfx.json` file:
```

Comment:
I think we should not mention `--force-gc`. This is only for testing purposes. Maybe we should instead mention last-resort GC calls `__motoko_gc_trigger()` as an expert option.